cmd/tailscaled/tailscaled.go

```diff
@@ -96,6 +97,70 @@ var subCommands = map[string]*func([]string) error{
 	"debug":                   &debugModeFunc,
 }
 
+// tryWindowsAppDataMigration attempts to move the Windows state storage
+// directory from its old location to the new location. (Issue 2856)
+//
+// Tailscale 1.14 and before stored state under %LocalAppData%
+// (usually "C:\WINDOWS\system32\config\systemprofile\AppData\Local"
+// when tailscaled.exe is running as a non-user system service).
+// However it is frequently cleared for almost any reason: Windows
+// updates, System Restore, even various System Cleaner utilities.
+//
+// This function does not return an error; on rename failure, code elsewhere
+// in Tailscale will continue to use the old location as a fallback.
+func tryWindowsAppDataMigration() {
+	if runtime.GOOS != "windows" {
+		return
+	}
+
+	oldFile := filepath.Join(os.Getenv("LocalAppData"), "Tailscale", "server-state.conf")
+	_, err := os.Stat(oldFile)
+	if os.IsNotExist(err) {
+		// Common case for a freshly installed system.
+		return
+	}
+	if err != nil {
+		log.Printf("tryWindowsAppDataMigration failed; AppDataLocal: %v", err)
+		return
+	}
+
+	newFile := filepath.Join(os.Getenv("ProgramData"), "Tailscale", "server-state.conf")
+	_, err = os.Stat(newFile)
+	if err == nil {
+		// Common case for a system which was upgraded from 1.14, we copied the file
+		// to %PROGRAMDATA% while leaving the original in %LOCALAPPDATA%.
+		return
+	}
+	if !os.IsNotExist(err) {
+		log.Printf("tryWindowsAppDataMigration failed; new file: %v", err)
+		return
+	}
+
+	// copy server-state.conf from %LOCALAPPDATA% to %PROGRAMDATA%
+	contents, err := os.ReadFile(oldFile)
+	if err != nil {
+		log.Printf("tryWindowsAppDataMigration failed; read oldFile: %v", err)
+		return
+	}
+
+	os.MkdirAll(filepath.Dir(newFile), 0755)
+	err = os.WriteFile(newFile, contents, 0600)
+	if err != nil {
+		remove_err := os.Remove(newFile)
+		if remove_err == nil {
+			log.Printf("tryWindowsAppDataMigration failed; write newFile: %v", err)
+			return
```

Comment:
move these two returns out into one at the end of the "if removeErr" block, so it's obvious that it doesn't fall through? and saves a line.

Reply:
Done